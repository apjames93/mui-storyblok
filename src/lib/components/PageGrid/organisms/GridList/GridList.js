/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';

import {
  GridList as MuiGridList,
  GridListTile as MuiGridListTile,
} from '@material-ui/core';

import StoryBlok from 'lib/utils/Storyblok';
import {
  validComponents,
  muiDimensionProp,
} from 'lib/utils/customProps';
import GridListTileBar from './components/GridListTileBar/GridListTileBar';

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
    overflow: 'hidden',
    backgroundColor: theme.palette.background.paper,
  },
  icon: {
    color: 'rgba(255, 255, 255, 0.54)',
  },
}));

const GridList = ({
  cellHeight,
  content,
  cols,
  spacing,
  rootClass,
  width,
  height,
  dataBlokC,
  dataBlokUid,
  storyblokClass,
}) => {
  const classes = useStyles();
  const styles = StoryBlok.arrayToMuiStyles(rootClass, { width: +width, height: +height });

  return (
    <div className={classes.root}>
      <MuiGridList
        cols={+cols}
        cellHeight={+cellHeight}
        spacing={+spacing}
        className={`${styles.root} ${storyblokClass}`}
        data-blok-c={dataBlokC}
        data-blok-uid={dataBlokUid}
      >
        {content.map(tile => (
          <MuiGridListTile
            key={tile.image}
            cols={+tile.cols}
            rows={+tile.rows}
            className={tile.storyblokClass}
            data-blok-c={tile.dataBlokC}
            data-blok-uid={tile.dataBlokUid}
          >
            <img src={tile.image} alt={tile.title} />
            {tile.gridListTileBar.length !== 0 ? (<GridListTileBar {...tile.gridListTileBar[0]} />) : null}
          </MuiGridListTile>
        ))}
      </MuiGridList>
    </div>
  );
};

export default GridList;

GridList.propTypes = {
  /**
   * storyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component. See CSS API below for more details.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui prop: number | 'auto'
   * Number of px for one cell height.
   * You can set 'auto' if you want to let the children determine the height.
   */
  cellHeight: PropTypes.string,
  /**
   * mui prop: number
   * Number of columns.
   */
  cols: PropTypes.string,
  /**
   * mui prop: number
   * Number of px for the spacing between tiles.
   */
  spacing: PropTypes.string,
  /**
   * mui prop: number
   * Number of px for the spacing between tiles.
   */
  height(props, propName, componentName) {
    return muiDimensionProp(props, propName, componentName);
  },
  /**
   * mui prop: number
   * Number of px for the spacing between tiles.
   */
  width(props, propName, componentName) {
    return muiDimensionProp(props, propName, componentName);
  },
  /**
   * Content passed to render
   * components: MuiTitleData
   */
  content(props, propName, componentName) {
    const components = ['MuiTitleData'];
    return validComponents(props, propName, componentName, components);
  },
};

GridList.defaultProps = {
  rootClass: [],
  cellHeight: 180,
  cols: 2,
  spacing: 4,
  height: 500,
  width: 450,
  content: [],
};
