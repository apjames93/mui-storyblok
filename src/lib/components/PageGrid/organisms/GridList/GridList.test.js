import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import GridList from './GridList';

function setup() {
  const props = {
    cellHeight: '180',
    cols: '3',
    spacing: '4',
    width: '400',
    height: '500',
    content: [
      {
        gridlistTileBar: [
          {
            actionIcon: [
              {
                component: 'IconButtonRedirect',
                iconName: 'home',
              },
            ],
            component: 'GridListTileBar',
            subtitle: 'test1',
            title: 'home',
            titlePosition: 'bottom',
          },
        ],
        cols: '1',
        component: 'TileData',
        image: 'www.test.com',
        rows: '1',
        title: 'bubble gum',
      },
    ],
  };

  const comp = shallow(<GridList {...props} />);
  return { comp, props };
}

describe('<GridList />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<GridList {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render GridList', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });
});
