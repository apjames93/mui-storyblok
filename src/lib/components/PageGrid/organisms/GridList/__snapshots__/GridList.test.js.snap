// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridList /> snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <ul
    className="MuiGridList-root makeStyles-root-4"
    style={
      Object {
        "margin": -2,
      }
    }
  >
    <li
      className="MuiGridListTile-root"
      style={
        Object {
          "height": 184,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }
    >
      <div
        className="MuiGridListTile-tile"
      >
        <img
          alt="bubble gum"
          src="www.test.com"
        />
        <div
          className="MuiGridListTileBar-root makeStyles-root-5 MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
        >
          <div
            className="MuiGridListTileBar-titleWrap MuiGridListTileBar-titleWrapActionPosRight"
          >
            <div
              className="MuiGridListTileBar-title"
            >
              home
            </div>
            <div
              className="MuiGridListTileBar-subtitle"
            >
              <span>
                test1
              </span>
            </div>
          </div>
          <div
            className="MuiGridListTileBar-actionIcon"
          />
        </div>
      </div>
    </li>
  </ul>
</div>
`;
